src/lobsterpy/featurize/core.py

```diff
@@ -1146,3 +1132,530 @@ def get_fingerprint_df(
+
+        # Create a temp dict for faster lookup
+        icoxx_dict = {
+            (tuple(icoxx_p_d_t[0]), tuple(icoxx_p_d_t[2])): (icoxx_p_d_t[1], icoxxs[idx])
+            for idx, icoxx_p_d_t in enumerate(input_icoxx_list)
+        }
+        # Initialize an empty list to store the data that goes in BWDF computation
+        mapped_icoxx_data, missing_interactions = [], []
+
+        # Check if interaction exists in both rdf data / icoxx data, then add to complete data
+        for rdf_p_d_t in ref_rdf_data:
+            pair_translation_key = (tuple(rdf_p_d_t[0]), tuple(rdf_p_d_t[2]))
+            bond_length = rdf_p_d_t[1]
+            if pair_translation_key in icoxx_dict:
+                icoxx_bond_length, icoxx_value = icoxx_dict[pair_translation_key]
+                if round(np.absolute(bond_length - icoxx_bond_length), 5) <= 1e-5:
+                    complete_entry = (*rdf_p_d_t, icoxx_value)
+                    if complete_entry not in mapped_icoxx_data:
+                        mapped_icoxx_data.append(complete_entry)
+                else:
+                    missing_interactions.append(rdf_p_d_t)
+            else:
+                missing_interactions.append(rdf_p_d_t)
+
+        # Check if the missing interactions are reverse images in ref neighbours data which LOBSTER sometimes eliminates
+        for rdf_p_d_t in missing_interactions[:]:
+            reversed_translation = tuple(-1 * np.array(rdf_p_d_t[2]))
+            reversed_pair = tuple(reversed(rdf_p_d_t[0]))
+            reversed_key = (reversed_pair, reversed_translation)
+
+            if reversed_key in icoxx_dict:
+                icoxx_bond_length, icoxx_value = icoxx_dict[reversed_key]
+                if round(np.absolute(rdf_p_d_t[1] - icoxx_bond_length), 5) <= 1e-5:
+                    complete_entry = (*rdf_p_d_t, icoxx_value)
+                    if complete_entry not in mapped_icoxx_data:
+                        mapped_icoxx_data.append((*rdf_p_d_t, icoxx_value))
+                        missing_interactions.remove(rdf_p_d_t)
+
+        # Compute histogram of bond lengths from the reference rdf and
+        # icoxx data to get wasserstein distance
+        rdf_dist = [entry[1] for entry in ref_rdf_data]
+        icoxx_dist = [entry[1] for entry in mapped_icoxx_data]
+        rdf_hist, _ = np.histogram(
+            rdf_dist,
+            bins=np.arange(0, self.max_length + self.bin_width, self.bin_width),
+            density=False,
+        )
+        icoxx_hist, _ = np.histogram(
+            icoxx_dist,
+            bins=np.arange(0, self.max_length + self.bin_width, self.bin_width),
+            density=False,
+        )
+
+        return {
+            "ref_rdf_data": ref_rdf_data,
+            "input_icoxx_list": input_icoxx_list,
+            "mapped_icoxx_data": mapped_icoxx_data,
+            "missing_interactions": missing_interactions,
+            "wasserstein_dist_to_rdf": wasserstein_distance(rdf_hist, icoxx_hist),
```

Comment:
Is this documented somewhere? I don't think I can see a description .

Reply:
Added detailed descriptions now in doc-strings. Thanks for catching it